source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/NumberInputHandler.java

```diff
@@ -43,14 +43,20 @@ public boolean isValidOnSpecifics(String numberInputValue)
             return false;
         }
 
+        // Before performing any validation, if the input value is empty and is not required, then it's valid
+        if (numberInputValue.isEmpty() && !numberInput.GetIsRequired())
+        {
+            return true;
+        }
+
         int inputValue = 0;
         try
         {
             inputValue = Integer.parseInt(numberInputValue);
         }
         catch (Exception ex)
         {
-            // Something failed, don't consider valid
+            // Parsing failed, consider it invalid
```

Comment:
What's our logging story in Java? Ideally this would get logged to stdout or the Debug stream for debugging etc..

Reply:
I don't think it should be logged, this will happen if an user writes something that is not a number into the input field

Reply:
True, maybe the catch could be narrowed to `NumberFormatException`.